File: API/Features/Reservations/Dtos/ReservationWriteDto.cs
using System;
using System.Collections.Generic;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations {

    public class ReservationWriteDto : IMetadata {

        public Guid ReservationId { get; set; }
        public int PaymentStatusId { get; set; }
        public string FromDate { get; set; }
        public string ToDate { get; set; }
        public string Remarks { get; set; }
        public string FinancialRemarks { get; set; }
        public bool IsDocked { get; set; }
        public bool IsLongTerm { get; set; }
        public bool IsAthenian { get; set; }
        public ReservationBoatDetailsDto Boat { get; set; }
        public ReservationInsuranceDetailsDto Insurance { get; set; }
        public ReservationOwnerDetailsDto Owner { get; set; }
        public ReservationBillingDetailsDto Billing { get; set; }
        public ReservationFeeDetailsDto Fee { get; set; }
        public List<ReservationBerthWriteDto> Berths { get; set; }
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/reservations/classes/dtos/reservation-write-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { BerthWriteDto } from './berth-write-dto'
import { ReservationBoatWriteDto } from './reservation-boat-write-dto'
import { ReservationFeeDto } from './reservation-fee-dto'
import { ReservationInsuranceDto } from './reservation-insurance-dto'
import { ReservationPersonDto } from './reservation-person-dto'
import { ReservationFishingLicenceDto } from './reservation-fishing-licence-dto'

export interface ReservationWriteDto {

    reservationId: Guid
    fromDate: string
    toDate: string
    remarks: string
    financialRemarks: string
    isAthenian: boolean
    isDocked: boolean
    isDryDock: boolean
    isRequest: boolean
    paymentStatusId: number
    berths: BerthWriteDto[]
    boat: ReservationBoatWriteDto
    fishingLicence: ReservationFishingLicenceDto
    insurance: ReservationInsuranceDto
    owner: ReservationPersonDto
    billing: ReservationPersonDto
    fee: ReservationFeeDto
    putAt: string

}
